src/redux/books/books.js

```diff
@@ -1,54 +1,42 @@
-import { ADD_NEW_BOOK, REMOVE_BOOK, GET_BOOKS_API } from '../types';
-import ApiServices from '../../services/Api.services';
+import { apiAddBook, apiGetBooks, apiRemoveBook } from '../../modules/api';
 
-const defualtBooks = [];
+const ADD = 'bookstore/Book/ADD';
+const GET = 'bookstore/Book/GET';
+const REMOVE = 'bookstore/Book/REMOVE';
 
-export function getBooksAPI(books) {
-  const APIBooks = Object.entries(books).map(([key, value]) => ({ ...value[0], id: key }));
-  return {
-    type: GET_BOOKS_API,
-    payload: APIBooks,
-  };
-}
-
-export const bookFromAPI = () => async (dispatch) => {
-  const response = await ApiServices.getBooks();
-  dispatch(getBooksAPI(response));
+export const addBook = (payload) => async (dispatch) => {
+  const book = { ...payload, item_id: payload.id, category: 'category' };
+  await apiAddBook(book);
+  dispatch({ type: ADD, book });
 };
 
-export function addNewBook(newBook) {
-  return {
-    type: ADD_NEW_BOOK,
-    payload: newBook,
-  };
-}
-
-export const addNewBooks = (newBook) => async (dispatch) => {
-  await ApiServices.addBook(newBook);
-  dispatch(addNewBook({ ...newBook, id: newBook.item_id }));
+export const getBooks = () => async (dispatch) => {
+  const data = await apiGetBooks();
+  const books = Object.keys(data).map((key) => {
+    const book = data[key][0];
+    book.id = key;
+    return book;
+  });
+  dispatch({ type: GET, books });
 };
 
-export function removeBook(bookId) {
-  return {
-    type: REMOVE_BOOK,
-    payload: bookId,
-  };
-}
-
-export const removeBooks = (id) => async (dispatch) => {
-  await ApiServices.removeBook(id);
-  dispatch(removeBook(id));
+export const removeBook = (id) => async (dispatch) => {
+  await apiRemoveBook(id);
+  dispatch({ type: REMOVE, id });
 };
 
-export default function reducerBook(initialState = defualtBooks, action) {
+export default function reducer(state = [], action = {}) {
   switch (action.type) {
-    case ADD_NEW_BOOK:
-      return [...initialState, action.payload];
-    case REMOVE_BOOK:
-      return [...initialState.filter((book) => (book.id !== action.payload))];
-    case GET_BOOKS_API:
-      return [...action.payload];
+    case ADD:
+      return [...state, action.book];
+
+    case GET:
+      return action.books;
+
+    case REMOVE:
+      return [...state].filter((book) => book.id !== action.id);
```

Comment:
- [ ] As requested by the previous reviewer and as mentioned in the project requirements kindly use `createAsyncThunk`  in the project for creating thunk action creators 👍